### PR TITLE
eBPF Flow documentation

### PR DESCRIPTION
**Pull request by andrekassis**
eBPF Documentation

### CODE REVIEW

**Comment by safchain** [comment]
Thanks @andrekassis it looks really nice. I would just make it a blog post instead of being part of the documentation as you are more describing testing, benchmark, approaches, future work and conclusion.

BTW the header of the file is missing, that is why jekyll complains. You can have a look here for example
https://github.com/skydive-project/skydive.network/tree/master/blog

File: documentation/skydive-with-ebpf.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+# Evaluation
+
+As mentioned above, the main challenge that the novel eBPF approach faces is the number of concurrent connections. In fact, it is almost agnostic to the rate at which new packets arrive, as opposed to other packet probes such as PCAP, AFPACKET, etc...
+
+Hence, our test gradually increments the number of flows to detect the system&#39;s breaking point. The number of parallel connections ranges from 1 to 1M.
+
+####               Settings
```
Comment (safchain) [inline documentation/skydive-with-ebpf.md]: nit: remove spaces

Comment (andrekassis) [reply documentation/skydive-with-ebpf.md]: done

**Comment by andrekassis** [comment]
All necessary changes were made @safchain

File: blog/skydive-with-ebpf.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+# Evaluation
+
+As mentioned above, the main challenge that the novel eBPF approach faces is the number of concurrent connections. In fact, it is almost agnostic to the rate at which new packets arrive, as opposed to other packet probes such as PCAP, AFPACKET, etc...
+
+Hence, our test gradually increments the number of flows to detect the system&#39;s breaking point. The number of parallel connections ranges from 1 to 1M.
+
+#### Settings
```
Comment (safchain) [inline blog/skydive-with-ebpf.md]: just `###` to make it more readable

File: blog/skydive-with-ebpf.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+1. We use a packet generator provided by Sylvain Afchain ([https://gist.github.com/safchain/b86b9a6a42e7fcd17f6df1a0ca827ae3](https://gist.github.com/safchain/b86b9a6a42e7fcd17f6df1a0ca827ae3)) to generate packets from one endpoint to another.
+2. The sender is placed in one host wholly isolated from the receiver (i.e., where Skydive is deployed)
+3. Skydive runs in two different pods, one for the agent and the other for the analyzer.
+4. We measure the CPU and memory consumption while the stress tests are running.
+5. We use the packet generator to set the number of flows on which the packets are sent and examine the effects this parameter has on the system.
+6. The test generates packets of 1024 bytes each (minimal, to maximize the number of context switches and simulate more severe stress).
+7. The only variable we change from one run to another is the number of connections.
+
+#### Results &amp; Analysis
```
Comment (safchain) [inline blog/skydive-with-ebpf.md]: same here

File: blog/index.md [hunk 4]
```diff
@@ -73,3 +73,8 @@ During the last few weekds, we added a TripleO job to our CI. In this blog post,
 ## by Nicolas Planel, 24/07/2018
 
 Skydive Flow Matrix is a tool on top of Skydive that helps you understand which services are connecting to each other on your platform. Thanks to the Skydive SocketInfo probe, Flow Matrix will report all opened Sockets between client and server processes across hosts...
+
+# [Skydive with eBPF](/blog/skydive-with-ebpf.html)
```
Comment (safchain) [inline blog/index.md]: move this block at the beginning of top of the list so that we get last written first

File: blog/skydive-with-ebpf.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+- One last important point to address in the future is eBPF&#39;s inability to provide data for specific protocols such as DNS, DHCP, etc… In order to provide application level information, the payload of each packet may need to be copied to the user-space, which is what this specific approach strives to eliminate.
+
+# How to Use
+
+To learn how to deploy Skydive and start an eBPF capture, please refer to [http://skydive.network](http://skydive.network). You can find the specific instructions on starting new captures at [http://skydive.network/tutorials/first-steps-6.html](http://skydive.network/tutorials/first-steps-6.html). To start a new capture, follow the instructions available at this above link – the relevant command line would be:
+
+skydive client capture create --gremlin &quot;G.V().Has(&#39;Name&#39;, &#39;eth0&#39;)&quot;
```
Comment (safchain) [inline blog/skydive-with-ebpf.md]:
```
{% highlight shell %}
<command line>
{% endhighlight %}
```

File: blog/skydive-with-ebpf.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+However, in order to make it work with the eBPF probe and not the default PCAP, the command line would need to be extended as follows:
+
+skydive client capture create --gremlin &quot;G.V().Has(&#39;Name&#39;, &#39;eth0&#39;)&quot; –-type &quot;ebpf&quot;
```
Comment (safchain) [inline blog/skydive-with-ebpf.md]: same here

**Comment by safchain** [comment]
can you amend you commit too with the right commiter and a better description, thanks :)

**Pull request merged**